docs: landing page redesign

montogeek opened the pull request:
## Summary

Implements new landing page as discussed in #7265

## Test plan

Run website locally, make sure it looks as the design. Preview: https://deploy-preview-7566--jest-preview.netlify.com/

Dec 30 progress:
- 70% of the homepage implemented using new design.

ToDo:

- [x] Cards section
- [x] Main information (white text over gray background)
- [x] Yellow bullet points
- [x] Philosophy

Notes:
Docusaurus fulfills its purpose, but its CSS architecture leaves much to be desired, making really hard to make some changes, also it doesn't support custom Navigation bar, only `Footer`, I am expressing these concerns because you will find some ugly CSS that could be completely avoided if Docusaurus was different.

SimenB commented:
So cool! It's deployed here: https://deploy-preview-7566--jest-preview.netlify.com/

>Docusaurus fulfills its purpose, but its CSS architecture leaves much to be desired, making really hard to make some changes

Will be improved in the next version: https://docusaurus.io/blog/2018/09/11/Towards-Docusaurus-2#layout

Not sure if an alpha is published, or viable to use?

SimenB commented on website/pages/en/index.js:
```diff
@@ -414,51 +369,67 @@ class Index extends React.Component {
+                  <MarkdownBlock>
+                    <translate>
+                      A lot of people! With
+                      [8.5m](https://www.npmjs.com/package/jest) downloads in
```
any way of keeping these numbers updated automatically?

Might be a cool docusaurus feature (showing stats from npm and github), /cc @endiliey

montogeek replied:
Yep, that would be really useful

montogeek commented:
Do'h, I forgot to uncomment my progress :/

montogeek commented:
> Will be improved in the next version: https://docusaurus.io/blog/2018/09/11/Towards-Docusaurus-2#layout

That's good to know, It is there any progress or proposition about how exactly it is going to be?

orta commented:
I think to one way to simplify the header CSS, is to make the custom nav CSS only affect the index page - then the rest of the pages can use the navigation bar etc. Which at least scopes the hacks to a single page.

orta commented:
We have a working implementation of the cards in #7567 - so I've ticked that for you 👍

cpojer commented:
I think Jest's philosophy hasn't changed much over the years:
* Fast
* Feature-rich
* Easy to use out of the box but highly configurable
* Approachable with familiar APIs
* Extensible
* Well documented
* Fun!

How about:
> Jest is a JavaScript testing framework designed to ensure correctness of any JavaScript codebase. It allows you to write tests with its approachable, familiar and feature-rich APIs that give you results quickly. Jest is extensively documented, requires little configuration and can be extended to match your requirements. Jest makes testing *delightful*.

?

cpojer commented:
The cards are fantastic. I have some thoughts on how I think it could lead to a better first impression:
* Can we make the initial animation a bit faster? Like, let's turn one card around within ~1s of opening the page. Otherwise we make it seem like Jest is slow.
* I'd like for at least 2 cards to be green at all times. It could be 2, 3 or 4 green cards, and at most 2 red cards. It doesn't give a good impression if the cards "fail" on the first visit to the website.

montogeek commented:
I did the yellow points (features) section.
Will finish last 2 things tomorrow.

SimenB commented:
@montogeek ping 🙂

montogeek commented:
@SimenB Was sick this week, will try today. Thanks for the ping.

montogeek commented:
I just completed latest 2 sections, there is a lot of details to tidyup.
Please review it and post your feedback.

SimenB commented:
Awesome, thank you! 🎉 

Mind fixing the linting error and the conflict?

---

Looked through the site now (didn't look at the code), and these are the things I noticed (overall it looks really good, though):

The colors in the header is off on mobile:

![image](https://user-images.githubusercontent.com/1404810/51004688-06558600-153c-11e9-8f99-e6d5a2c312bd.png)

GitHub widget is floating to the right. I know it's in the sketches from @orta, but I think it looks a bit weird...

![image](https://user-images.githubusercontent.com/1404810/51004790-93004400-153c-11e9-9867-a97765b3b450.png)

Maybe drop to upper case `W` in `With` in the tagline?

![image](https://user-images.githubusercontent.com/1404810/51004778-7cf28380-153c-11e9-9f01-bb2b8314f308.png)

Could we have `it` and `expect` as monospace (`code` or `pre`, I guess? I don't remember my semantics 😅)

![image](https://user-images.githubusercontent.com/1404810/51004800-9c89ac00-153c-11e9-8b00-b98b570adac4.png)

Still some lorum ipsum in the talks section, I guess somebody:tm: needs to figure out some text

![image](https://user-images.githubusercontent.com/1404810/51004863-ce027780-153c-11e9-9544-0509f14c9cdf.png)

Can we do something about the background behind sponsors, a couple of the logos are not legible:

![image](https://user-images.githubusercontent.com/1404810/51004727-33a23400-153c-11e9-9123-7c54bfcc1350.png)

Should the company logos be left-aligned to match open collective?

![image](https://user-images.githubusercontent.com/1404810/51004739-4583d700-153c-11e9-92b7-1f57b6f894c0.png)

montogeek commented:
1. Lint and conflict fixed.
2. Navbar colors on mobile fixed.
3. GitHub widget is not rendered on mobile, on desktop, for me looks fine, final decision is on you and maintainers.
4. With uppercase fixed.
5. Fixed, added backticks.
6. Those logos are PNG with transparent background, I added a `grey` background.
7. Company logos are actually left aligned. Sorry, not sure what you mean.

montogeek commented:
@SimenB Thanks for the review, please check it again.

SimenB commented on website/static/css/jest.css:
```diff
@@ -1,9 +1,26 @@
+  --grey3: #545454;
+}
+
+.debug * {
```
remove?

montogeek replied:
Just did

SimenB commented:
Note, it'd be cool to hold off on merging until we release Jest 24. Should be any day now (and I can handle any conflicts) 🙂

So if we get the blog post ready, we can merge that to this branch, then land both at the same time

SimenB commented:
One more, can you make the links brake to 2+2 on mobile?

thymikee commented:
Great seeing this shaping up! I have some feedback as well, so here it is

On mobile/tablet view, text should be white when hover/focus nav items

<img width="535" alt="screenshot 2019-01-12 at 20 38 27" src="https://user-images.githubusercontent.com/5106466/51077716-84379f80-16aa-11e9-962b-856072c2c128.png">

How do we feel about adding slight shadow to the nav on mobile/tablet view to make it distinct from the content? It would actually also make sense on the docs pages on desktop as well

<img width="888" alt="screenshot 2019-01-12 at 20 40 34" src="https://user-images.githubusercontent.com/5106466/51077717-84379f80-16aa-11e9-9960-31e09f4f0df5.png">

Seems just odd. Also github stars is not clickable

<img width="154" alt="screenshot 2019-01-12 at 20 42 21" src="https://user-images.githubusercontent.com/5106466/51077718-84379f80-16aa-11e9-8600-99cbd3bdc298.png">

Because "language selection" is a span, it doesn't flex properly

<img width="667" alt="screenshot 2019-01-12 at 20 46 36" src="https://user-images.githubusercontent.com/5106466/51077775-29527800-16ab-11e9-8571-758f5100717c.png">

Welp. Btw, maybe it's right time to move language selection next to logo and version? Having language as a regular nav property next to "API" always felt weird.

<img width="382" alt="screenshot 2019-01-12 at 20 47 53" src="https://user-images.githubusercontent.com/5106466/51077790-4ab36400-16ab-11e9-9dc8-6f4a32f588fd.png">

How do we feel about centring this tagline?

<img width="1058" alt="screenshot 2019-01-12 at 20 51 23" src="https://user-images.githubusercontent.com/5106466/51077831-e80e9800-16ab-11e9-8739-26edf38a51b9.png">

Speaking of which, this is quite a convoluted markup out there, can we remove these unnecessary divs and spans, empty paragraphs?

<img width="364" alt="screenshot 2019-01-12 at 20 52 20" src="https://user-images.githubusercontent.com/5106466/51077832-e80e9800-16ab-11e9-9a90-f769ffb2f732.png">

brainkim commented on website/pages/en/index.js:
```diff
@@ -113,90 +103,155 @@ class Contributors extends React.Component {
+    return (
+      <div className="jest-hand">
+        {cards}
+        <script src="landing.js" />
```
```suggestion
        <script src="/landing.js" />
```
Cards are broken with i18n urls (https://deploy-preview-7566--jest-preview.netlify.com/ro/) because it looks for the script in the wrong place.
Thanks for doing this!

orta commented:
Hey folks, I've moved this bit closer ⭐️ - it's really starting to feel alive now

> re: Dom elements

These come from the docsaurus / translation APIs, so we can't really change those.

I like the idea of moving the langauge selector over to the left side  is a good call, they fit👍 

--- 

Mobile support is still something we'll need to think about, but it's all looking good

SimenB commented:
Mobile is really broken after the latest updates

![image](https://user-images.githubusercontent.com/1404810/51090552-861e6300-177d-11e9-931a-158bf0fb2c28.png)

cpojer commented:
Man, this is really nicely coming together. The green Jest logo looks much better imho, thanks for doing that. Here are some thoughts:

* Having both an intro and the philosophy separate seems a bit much. Could we collapse the intro and philosophy into one block?
* The hover state for links in the menu doesn't work well with this redesign. The language button hover state touches the search bar. Could we add some margin between items? Also, could we get rid of the fade or speed it up? It looks jarring.
* Could we turn the color of the Jest card to green as well?
* There should be some subtle hover state for the buttons below the cards
* The screenshots should probably be clickable so they can be viewed in bigger size?
* The GitHub "star" button doesn't work.
* The philosophy (my fault) has "extend" and "extensively" which sounds repetitive: "Jest is extensively documented, requires little configuration and can be extended to match your requirements. Jest makes testing delightful." Could we change it to "Jest is well-documented, requires little configuration and can be extended to match your requirements. Jest makes testing delightful."?

jamesgeorge007 commented:
@SimenB I would like to on the mobile responsiveness part after this one gets merged :+1:

jamesgeorge007 commented:
@montogeek Squashing these commits into one would be a good idea.

orta commented:
I've got this considerably closer, and plan t get back to it today - mobile is now first class again.

I've switched the front-most card to be green. Warning, it takes a bit of getting used to seeing the jester shoe in green (I explored having it in grey, but didn't really work out, and keeping it in the old color didn't mesh well with the card name. Maybe a really washed out red _could work_ )

@cpojer I changed the wording a tiny bit in the philosophy section, mainly to make it flow a bit better and added your improvements

SimenB commented:
There's some weird indent by "zero config"

![image](https://user-images.githubusercontent.com/1404810/51431465-506bf500-1c29-11e9-85dd-089fa4872d5b.png)

---

The color uses in the headings are a bit all over, maybe on purpose? It's yellow, red, yellow, red, green, green, yellow, yellow

---

Could we move the citation under the philosophy so it takes up full width:

![image](https://user-images.githubusercontent.com/1404810/51431478-a6409d00-1c29-11e9-8a8b-090dc8481c2b.png)

---

Should the buttons align with the text?

![image](https://user-images.githubusercontent.com/1404810/51431490-beb0b780-1c29-11e9-8c21-7dcb88c478c4.png)

---

Hover on buttons in the menu still look a bit weird, might just be me?

![image](https://user-images.githubusercontent.com/1404810/51431498-dbe58600-1c29-11e9-8a54-58db81ae38e4.png)

thymikee commented:
I'm really not sure about changing Jest logo to green. I'm cool with [JEST] logo-label being green indicating success, but I'd just leave the actual logo color and website theme as it was – burgundish is not really bright red so at least my brain doesn't associate it with a failure.

orta commented:
Ah, sorry, I forgot to commit making the green darker last night - it's much better now

orta commented:
That's almost all feedback fixed, and IMO it's pretty food to go.

brainkim commented:
Psst. Just mentioning that the cards are broken when you set the locale in the url b/c we need an absolute link not a relative one here: https://github.com/facebook/jest/pull/7566#discussion_r247329339 🙆‍♀️

orta commented:
Thanks for the reminder, fixed!

cpojer commented:
* I think adding a subtle hover style for the top bar (`opacity: 80%` or something) and `text-decoration: none` for regular green links may be a good idea.
* I'd add a full-stop `.` at the end of the sentence "Jest is a delightful JavaScript Testing Framework with a focus on simplicity".
* Since the "JEST" logo in the top bar changed, it may be best to revert the change from my earlier recommendation of reducing font-size of the version number. I think it would look better now if the font-size was the same as the JEST text again.

Other than those minor things, I love it. Shipit!

orta commented:
OK, I've shipped all of those changes too 👍

SimenB commented:
(I still think it would be cool to land together with the blog post for Jest 24, feel free to ignore if you disagree, though)

cpojer commented:
Thanks for making those changes but seems like the green text doesn't have a hover style still. I agree with @SimenB it would be awesome to land together; and mention it in the blog post, but if you think we should do it separately I'm ok with that too.

One thing that @gaearon brought up was not re-running the full animation on subsequent reloads. I'm wondering if we could disable the animation to only play once in an hour or once in a day; and possibly speed it up on every re-visit of the site, indicating that the "test run" gets faster each time. I think that would be a cool gimmick using localStorage. What do you think?

cpojer commented:
One more on the headlines:
* Excellent Exceptions -> "Great Exceptions"? (I get the pun but feel like it's overstating it a bit)
* Mocking with ease -> "Easy Mocking"?

orta commented:
Oops, missed the hover state - fixed.

> not re-running the full animation on subsequent reloads

When I tested this site on folks at our [Peer Labs](https://www.meetup.com/CocoaPods-NYC/) all but one immediately re-watched the animation was by reloading the page. I think disabling it on the second time would end up making that a bit disappointing (or feel buggy). 

So, I've made 2nd runs have half the base wait-time that the 1st run has 👍

cpojer commented:
You are amazing. Thanks for doing this.

SimenB commented:
Last 2 things, I swear

![image](https://user-images.githubusercontent.com/1404810/51531119-e035bd00-1e3c-11e9-8a92-58b4820b1c8f.png)

Can we increase the contrast of the placeholder and icon against the background? The screenshot looks OK since it's so zoomed in - on my screen (macbook pro 15") it's pretty hard to read

---

`/blog` has `.blog .wrapper { max-width: 1100px; }` which makes it more narrow than all other pages. Just removing that looks OK to me (the blog text will wrap due to the toc added on the right side, but it already moves to the left when opening it, so it changes now as well).
It looks pretty dumb that the header changes width, at least

normal:
![image](https://user-images.githubusercontent.com/1404810/51531274-47ec0800-1e3d-11e9-933e-c035709d5efb.png)

blog:
![image](https://user-images.githubusercontent.com/1404810/51531290-4e7a7f80-1e3d-11e9-8b52-5deafa34ede4.png)

montogeek commented:
Thanks!

orta commented:
@montogeek - you can now claim the bounty mentioned in https://github.com/facebook/jest/issues/7265#issuecomment-450507843 <3 thanks!

montogeek commented:
@orta Thanks to you and the team!
